lightning/src/util/ser.rs

```diff
@@ -262,15 +269,16 @@ impl<'a, T: Writeable> Writeable for VecWriteWrapper<'a, T> {
 }
 
 /// Wrapper to read elements from a given stream until it reaches the end of the stream.
-pub(crate) struct VecReadWrapper<T: Readable>(pub Vec<T>);
-impl<T: Readable> Readable for VecReadWrapper<T> {
+pub(crate) struct VecReadWrapper<T>(pub Vec<T>);
+impl<T: MaybeReadable> Readable for VecReadWrapper<T> {
```

Comment:
Not sure I understand the comment. Obviously this PR is a bit awkward because it is just commits from #1034, which does use all the code in it. I tried a few ways to do `VecReadWrapper` that I thought were cleaner but always ended up hitting the "you may implement `MaybeReadable` and `Readable` for this type causing duplicate implementations" stuff every time.

Reply:
No action necessary. Just pointing out this use case.